From 4a233ed7756049e4b5f2b1ea3b7fcfc19c6ddace Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Wed, 26 Aug 2020 22:24:11 -0700
Subject: [PATCH 1534/1851] v4l: xilinx: multi-scaler: Fix warnings for
 UNUSED_VALUE

commit dac779b7cba59cf96ea0ab8a5aa93a7cdf83e72f from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNUSED_VALUE type.

The actual warnings are:

- For xm2msc_select_hcoeff()
Assigning value from "chan_ctx->xm2msc_dev->taps" to "ntaps" here,
but that stored value is overwritten before it can be used.

Overwriting previous write to "ntaps" with value "12U".

- For xm2msc_select_vcoeff()
Assigning value from "chan_ctx->xm2msc_dev->taps" to "ntaps" here,
 but that stored value is overwritten before it can be used.

Overwriting previous write to "ntaps" with value "12U".

- For vidioc_s_fmt()
Assigning value from "get_q_data(chan_ctx, f->type)" to "q_data" here,
 but that stored value is overwritten before it can be used.

Overwriting previous write to "q_data" with value from
 "get_q_data(chan_ctx, f->type)".

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 3daf90d3377d..5cc2af8ac1e9 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -600,7 +600,7 @@ xm2msc_select_hcoeff(struct xm2msc_chan_ctx *chan_ctx, const short **coeff)
 	u16 hscale_ratio;
 	u32 width_in = chan_ctx->q_data[XM2MSC_CHAN_OUT].width;
 	u32 width_out = chan_ctx->q_data[XM2MSC_CHAN_CAP].width;
-	u32 ntaps = chan_ctx->xm2msc_dev->taps;
+	u32 ntaps;
 
 	if (width_out < width_in) {
 		hscale_ratio = (width_in * 10) / width_out;
@@ -664,7 +664,7 @@ xm2msc_select_vcoeff(struct xm2msc_chan_ctx *chan_ctx, const short **coeff)
 	u16 vscale_ratio;
 	u32 height_in = chan_ctx->q_data[XM2MSC_CHAN_OUT].height;
 	u32 height_out = chan_ctx->q_data[XM2MSC_CHAN_CAP].height;
-	u32 ntaps = chan_ctx->xm2msc_dev->taps;
+	u32 ntaps;
 
 	if (height_out < height_in) {
 		vscale_ratio = (height_in * 10) / height_out;
@@ -1570,7 +1570,7 @@ static int
 vidioc_s_fmt(struct xm2msc_chan_ctx *chan_ctx, struct v4l2_format *f)
 {
 	struct v4l2_pix_format_mplane *pix = &f->fmt.pix_mp;
-	struct xm2msc_q_data *q_data = get_q_data(chan_ctx, f->type);
+	struct xm2msc_q_data *q_data;
 	unsigned int i;
 	unsigned int align = 1;
 
-- 
2.31.1

